Source/Core/Core/HW/GBACore.cpp

```diff
@@ -0,0 +1,518 @@
+// Copyright 2021 Dolphin Emulator Project
+// Licensed under GPLv2+
+// Refer to the license.txt file included.
+
+#include "Core/HW/GBACore.h"
+
+#define PYCPARSE  // Remove static functions from the header
+#include <mgba/core/interface.h>
+#undef PYCPARSE
+#include <mgba-util/vfs.h>
+#include <mgba/core/blip_buf.h>
+#include <mgba/core/log.h>
+#include <mgba/core/timing.h>
+#include <mgba/internal/gba/gba.h>
+#include <unzip.h>
+
+#include "AudioCommon/AudioCommon.h"
+#include "Common/ChunkFile.h"
+#include "Common/CommonPaths.h"
+#include "Common/CommonTypes.h"
+#include "Common/Config/Config.h"
+#include "Common/FileUtil.h"
+#include "Common/MinizipUtil.h"
+#include "Common/Thread.h"
+#include "Core/Config/MainSettings.h"
+#include "Core/Core.h"
+#include "Core/HW/GBAFrontend.h"
+#include "Core/HW/GBAPad.h"
+#include "Core/HW/SI/SI_DeviceGCController.h"
+#include "Core/HW/SystemTimers.h"
+#include "Core/Movie.h"
+#include "Core/NetPlayProto.h"
+
+namespace HW::GBA
+{
+namespace
+{
+mLogger s_dummy_logger = {
+    [](mLogger*, int category, mLogLevel level, const char* format, va_list args) {}, nullptr};
+}  // namespace
+
+constexpr auto SAMPLES = 512;
+constexpr auto SAMPLE_RATE = 48000;
+
+static FrontendFactory s_frontend_factory = [](int, std::string_view, std::string_view, u32, u32) {
+  return std::make_unique<FrontendInterface>();
+};
+
+void SetFrontendFactory(FrontendFactory factory)
+{
+  s_frontend_factory = factory;
+}
+
+Core::Core(int device_number, u64 gc_ticks)
+    : m_device_number(device_number), m_last_gc_ticks(gc_ticks)
+{
+  mLogSetDefaultLogger(&s_dummy_logger);
+  m_core = mCoreCreate(mPlatform::mPLATFORM_GBA);
+  m_core->init(m_core);
+
+  mCoreInitConfig(m_core, "dolphin");
+  mCoreConfigSetValue(&m_core->config, "idleOptimization", "detect");
```

Comment:
This can be more efficient than the default ``"remove"`` in some games. I left it there as a way to potentially avoid stalling Dolphin or lower CPU usage